docs/source/faq.rst

```diff
@@ -617,3 +617,61 @@ will retry failed trials when a new trial starts to evaluate.
     )
 
     study = optuna.create_study(storage=storage)
+
+
+How can deal with permutation as a parameter?
+---------------------------------------------
+
+Although it is not straightforward to deal with combinatorial search spaces like permutations with existing API, there exists a convenient technique for handling them. 
+It involves re-parametrization of permutation search space of :math:`n` items as an independent :math:`n`-dimensional integer search space. 
+This technique is based on the concept of `Lehmer code <https://en.wikipedia.org/wiki/Lehmer_code>`_, which is also explained in `Vizier document <https://oss-vizier.readthedocs.io/en/latest/guides/user/search_spaces.html#combinatorial-reparamterization>`_.
+
+A Lehmer code of a sequence is the sequence of integers in the same size, whose :math:`i`-th entry denotes how many inversions the :math:`i`-th entry of the permutation has after itself.
+In other words, the :math:`i`-th entry of the Lehmer code represents the number of entries that are located after and are smaller than the :math:`i`-th entry of the original sequence.
+For instance, the Lehmer code of the permutation :math:`(3, 1, 4, 2, 0)` is :math:`(3, 1, 2, 1, 0)`.
+
+Not only does the Lehmer code provide a unique encoding of permutations into an integer space, but it also has some desirable properties.
+For example, the sum of a Lehmer code entries is equal to the minimum number of adjacent transpositions necessary to transform the corresponding permutation into the identity permutation. 
+Additionally, the lexicographical order of the encodings of two permutations is the same as that of the original sequence. 
+Therefore, Lehmer code preserves "closeness" among permutations in some sense, which is important for the optimization algorithm.
```

Comment:
Yes, it is good point. But the example below use `TPESampler` as sampler, and this algorithm does not use such a kind of "closeness" information in my understanding. One idea would be to add a note regarding this, but it is a weak opinion. What do you think?

Reply:
It does considers as long as the parameters are in int search space, since int parameters are sampled from gaussian mixture and then rounded.

https://github.com/optuna/optuna/blob/a031e95dd7c93266d54cf8e7f99a455519b4dd81/optuna/samplers/_tpe/probability_distributions.py#L62-L74

Reply:
I see. I overlooked this point. Thank you for your answer.